src/components/molecules/UiLoader/UiLoader.vue

```diff
@@ -89,56 +90,108 @@ const props = defineProps({
     default: 'div',
   },
   /**
-   * Use this props to pas transition name
+   * Use this props to set transition name
    */
-  transition: {
-    type: [String, Boolean] as PropType<string | boolean>,
-    default: false,
+  name: {
+    type: String,
+    default: 'v',
   },
   /**
-   * Use this props to start rendering final component before loading stops, so it's completely rendered
+   * Use this props to apply a transition on the initial render
    */
-  eagerLoadComponent: {
+  appear: {
     type: Boolean,
-    default: false,
+    default: true,
   },
 });
-const component = computed<LoaderComponent>(() => {
+const loaderEl = ref<HTMLElement | null>(null);
+const contentEl = ref<HTMLElement | null>(null);
+const loaderComponent = computed<LoaderComponent>(() => {
   const components: Record<LoaderType, LoaderComponent> = {
     spinner: UiLoaderSpinner,
     ellipsis: UiLoaderEllipsis,
     skeleton: UiLoaderSkeleton,
   };
   return components[props.type];
 });
-const outerComponent = computed<VNode>(() => (
-  props.transition
-    ? h(TransitionGroup, { name: props.transition as string })
-    : h((_, { slots }) => (slots.default as Slot)())
-));
+const animateEl = ref<'loader' | 'content'>(props.isLoading ? 'loader' : 'content');
```

Comment:
I’m not sure if I understand this part of UiLoader correctly, but wouldn’t it be easier to decrease the number of variables (`refs` and `computed` properties) and have just 1 `animateEl` computed property, which could depend on `props.isLoading`? So

```ts
const animateEl = computed<'loader' | 'content'>(() => props.isLoading ? 'loader' : 'content');
```

should do the job, right? :D